src/frequenz/client/electricity_trading/_client.py

```diff
@@ -91,6 +92,31 @@ def validate_decimal_places(value: Decimal, decimal_places: int, name: str) -> N
         ) from exc
 
 
+async def grpc_call_with_timeout(
+    call: Callable[..., Awaitable[Any]], *args: Any, timeout: float = 300, **kwargs: Any
```

Comment:
I think this should take a `timedelta | None` and the default value should be `None` here and in the other calls.  Otherwise it wouldn't be intuitive and not what you'd expect from a thin client.

Reply:
I agree about the `timedelta` but I don't think that the timeout should be `None` by default. Else we risk having the API stuck again (without giving any errors) like we did last week

Reply:
Well, the actor would set it when making calls to the client.  The actor can get the value to use from the config file.

The 300s comes from specific issues we noticed in the server and what we assume the actor needs.  We can't say that 300 is the right value in the long term.  It could become different even as soon as we go to production.

Reply:
> No the 300s was arbitrary just to have some timeout in the API requests.

This is just the thing. Defaults have to make sense, and we can't find a value that does, especially when we expect external parties to use it. Maybe we shouldn't add this here at all and keep it only in the actors? If, at some point, we get an SDK-like layer between the client and the actor, these problems would become much easier.

Maybe @llucax has an opinion?  Because we try to be consistent with all the API clients.

In the worst case, we go ahead with 300s as the default if it works for most cases.

For example, the default receiver buffer size of 50 is a bit arbitrary and has bitten me sometimes. It didn't occur to me for a long time that it was overflowing and needed a bigger buffer. But still, having 50 as a default has been very useful in most places. The situation here is very different because `None` as buffer size would have meant no buffer, whereas here `None` as timeout means infinite timeout.